### PR TITLE
PropertyHarmonization

### PR DESCRIPTION
**Pull request by jmirtsch**
Initial Property Harmonization

### CODE REVIEW

File: IFC4x3/Properties/d/CargoDeadWeight_2vVxFsGGb3QOAIX9UEyhga/DocProperty.xml [hunk 1]
```diff
@@ -1,3 +0,0 @@
-<?xml version="1.0" encoding="utf-8"?>
-<DocProperty xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" id="CargoDeadWeight_2vVxFsGGb3QOAIX9UEyhga" Name="CargoDeadWeight" UniqueId="b97fb3f6-4109-4369-8292-84978ef2baa4" PrimaryDataType="IfcMassMeasure" />
```
Comment (AlexBrad1eyCT) [inline IFC4x3/Properties/d/CargoDeadWeight_2vVxFsGGb3QOAIX9UEyhga/DocProperty.xml:2]: This should not be deleted and replaces 'DeadWeightTonnage' name was altered to remove implicitly stating the unit in the name.

File: IFC4x3/Properties/d/CargoDeadWeight_2vVxFsGGb3QOAIX9UEyhga/Documentation.md [hunk 2]
```diff
@@ -1 +0,0 @@
-Weight of (bulk) cargo carried
```
Comment (AlexBrad1eyCT) [inline IFC4x3/Properties/d/CargoDeadWeight_2vVxFsGGb3QOAIX9UEyhga/Documentation.md:1]: As Above is a replacement for 'DeadWeightTonnage' should not be deleted.

File: IFC4x3/Properties/m/MitigatedRiskLikelihood_2mh3cZyunB0ONymM2LqXn/DocProperty.xml [hunk 3]
```diff
@@ -1,5 +0,0 @@
-<?xml version="1.0" encoding="utf-8"?>
-<DocProperty xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" id="MitigatedRiskLikelihood_2mh3cZyunB0ONymM2LqXn" Name="MitigatedRiskLikelihood" UniqueId="b0ac39a3-f38c-4b01-85fc-c16095d21c7f" PropertyType="p_enumeratedvalue" PrimaryDataType="IfcLabel">
```
Comment (AlexBrad1eyCT) [inline IFC4x3/Properties/m/MitigatedRiskLikelihood_2mh3cZyunB0ONymM2LqXn/DocProperty.xml:2]: Accidental deletion please revert, this element is still referenced in the Pset_Risk will cause an error in IfcDoc.

File: IFC4x3/Properties/m/MitigatedRiskLikelihood_2mh3cZyunB0ONymM2LqXn/Documentation.md [hunk 4]
```diff
@@ -1 +0,0 @@
-Identifies the likelihood of the hazard given the planned mitigation.
```
Comment (AlexBrad1eyCT) [inline IFC4x3/Properties/m/MitigatedRiskLikelihood_2mh3cZyunB0ONymM2LqXn/Documentation.md:1]: Accidental deletion please revert, this element is still referenced in the Pset_Risk will cause an error in IfcDoc.

File: IFC4x3/Properties/n/NominalDepth_1shYs0qXWHuO00025QrE$V/DocProperty.xml [hunk 5]
```diff
@@ -1,5 +1,5 @@
 <?xml version="1.0" encoding="utf-8"?>
-<DocProperty xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" id="NominalDepth_1shYs0qXWHuO00025QrEV" Name="NominalDepth" UniqueId="1shYs0qXWHuO00025QrE$V" PrimaryDataType="IfcPositiveLengthMeasure">
+<DocProperty xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" Name="NominalDepth" UniqueId="1shYs0qXWHuO00025QrE$V" PrimaryDataType="IfcPositiveLengthMeasure">
 	<Localization>
 		<DocLocalization Locale="en" Name="Nominal Depth" />
 		<DocLocalization Locale="ja-JP" Name="深さ">
```
Comment (AlexBrad1eyCT) [inline IFC4x3/Properties/n/NominalDepth_1shYs0qXWHuO00025QrE$V/DocProperty.xml:5]: Can we transfer the localization to the harmonized keep element NominalDepth_0HZnE0qUaHuO00025QrEV

File: IFC4x3/Properties/n/NominalWidth_2bcxE0qSOHuO00025QrE$V/DocProperty.xml [hunk 6]
```diff
@@ -1,5 +1,5 @@
 <?xml version="1.0" encoding="utf-8"?>
-<DocProperty xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" id="NominalWidth_2bcxE0qSOHuO00025QrEV" Name="NominalWidth" UniqueId="2bcxE0qSOHuO00025QrE$V" PrimaryDataType="IfcPositiveLengthMeasure">
+<DocProperty xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" Name="NominalWidth" UniqueId="2bcxE0qSOHuO00025QrE$V" PrimaryDataType="IfcPositiveLengthMeasure">
 	<Localization>
 		<DocLocalization Locale="en" Name="Nominal Width" />
 		<DocLocalization Locale="ja-JP" Name="公称幅">
```
Comment (AlexBrad1eyCT) [inline IFC4x3/Properties/n/NominalWidth_2bcxE0qSOHuO00025QrE$V/DocProperty.xml:5]: As before can we keep the localization

File: IFC4x3/PropertyConstants/v/VASCO COSTA_1jQ7XYFLbCQggk_8EEnXx4.xml [hunk 7]
```diff
@@ -1,3 +0,0 @@
-<?xml version="1.0" encoding="utf-8"?>
-<DocPropertyConstant xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" id="VASCO_COSTA_1jQ7XYFLbCQggk_8EEnXx4" Name="VASCO COSTA" UniqueId="6d687862-3d59-4c6a-aaae-f8838ec61ec4" />
```
Comment (AlexBrad1eyCT) [inline IFC4x3/PropertyConstants/v/VASCO COSTA_1jQ7XYFLbCQggk_8EEnXx4.xml:2]: @jmirtsch  can you correct the name to VASCO_COSTA it is a required Enumeration within PEnum_AddedMassCoefficientMethod type

**Review by AlexBrad1eyCT — CHANGES_REQUESTED** [review]
Reviewed the update, just a few comments, accidental deletion of a risk Property, and a deletion of a P&W constant that had a grammar mistake.

File: IFC4x3/Properties/d/Depth_19jxXCHUr5VhPM7hPVeFYR/DocProperty.xml [hunk 8]
```diff
@@ -1,3 +1,3 @@
 <?xml version="1.0" encoding="utf-8"?>
-<DocProperty xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" id="Depth_19jxXCHUr5VhPM7hPVeFYR" Name="Depth" UniqueId="49b7b84c-45ed-457e-b656-1eb65fa0f89b" PrimaryDataType="IfcPositiveLengthMeasure" />
+<DocProperty xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" Name="Depth" UniqueId="49b7b84c-45ed-457e-b656-1eb65fa0f89b" PrimaryDataType="IfcPositiveLengthMeasure" />
```
Comment (SergejMuhic) [inline IFC4x3/Properties/d/Depth_19jxXCHUr5VhPM7hPVeFYR/DocProperty.xml:3]: Is Depth orphaned?

File: IFC4x3/PropertyConstants/v/VASCO COSTA_1jQ7XYFLbCQggk_8EEnXx4.xml (hunk 7 above)
Comment (SergejMuhic) [reply IFC4x3/PropertyConstants/v/VASCO COSTA_1jQ7XYFLbCQggk_8EEnXx4.xml:2]: This one apparently got completely deleted.

File: IFC4x3/Sections/Domain specific data schemas/Schemas/IfcRail/PropertySets/Pset_RailwayAlignmentCommon/DocPropertySet.xml [hunk 9]
```diff
@@ -1,8 +1,3 @@
 <?xml version="1.0" encoding="utf-8"?>
-<DocPropertySet xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" Name="Pset_RailwayAlignmentCommon" UniqueId="2dde4d86-50a3-497c-8beb-e98d4d424c34" ApplicableType="IfcAlignment">
-	<Properties>
-		<DocProperty xsi:nil="true" href="CantRotationAxis_1IburP5x9EXf88H5s5Xwyq" />
-		<DocProperty xsi:nil="true" href="VerticalReferenceAxis_2z2sJ89ur2jem1qknkDlcP" />
-	</Properties>
-</DocPropertySet>
+<DocPropertySet xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" Name="Pset_RailwayAlignmentCommon" UniqueId="2dde4d86-50a3-497c-8beb-e98d4d424c34" ApplicableType="IfcAlignment" />
```
Comment (SergejMuhic) [inline IFC4x3/Sections/Domain specific data schemas/Schemas/IfcRail/PropertySets/Pset_RailwayAlignmentCommon/DocPropertySet.xml:2]: This pset needs to be removed completely Jon. Alex removed it in his PR already. I guess you did not have the latest baseline.

**Review by SergejMuhic — COMMENTED** [review]
Not sure if it was intended to have the orphans (probably makes sense to have a base for updating ifcdoc) and the changes in GUIDs. Those are in general my only concern (apart from the comment on the alignment pset).

File: IFC4x3/PropertyConstants/v/VASCO COSTA_1jQ7XYFLbCQggk_8EEnXx4.xml (hunk 7 above)
Comment (AlexBrad1eyCT) [reply IFC4x3/PropertyConstants/v/VASCO COSTA_1jQ7XYFLbCQggk_8EEnXx4.xml:2]: Yes it needs reverting and only the name being corrected

**Comment by AlexBrad1eyCT** [comment]
@SergejMuhic, I ignored the orphans as assume the scripting you want to try will remove them at some point,

@jmirtsch, are you able to sort the few changes asap so we can have a clean baseline for further rail harmonization. (not being pushy i promise!)

File: IFC4x3/Properties/d/CargoDeadWeight_2vVxFsGGb3QOAIX9UEyhga/DocProperty.xml (hunk 1 above)
Comment (jmirtsch) [reply IFC4x3/Properties/d/CargoDeadWeight_2vVxFsGGb3QOAIX9UEyhga/DocProperty.xml]: I hadn't noticed this change, but I should have and raised it.  This property is listed twice in the baseline I branched from, one in the d folder and one in c
https://github.com/bSI-InfraRoom/IFC-Specification/tree/64f1fc0e8cdf8aba82d2b99a423ddefae2f5a2ba/IFC4x3/Properties/d/CargoDeadWeight_2vVxFsGGb3QOAIX9UEyhga

File: IFC4x3/Properties/d/CargoDeadWeight_2vVxFsGGb3QOAIX9UEyhga/Documentation.md (hunk 2 above)
Comment (jmirtsch) [reply IFC4x3/Properties/d/CargoDeadWeight_2vVxFsGGb3QOAIX9UEyhga/Documentation.md]: Refer above, this file was duplicated in d folder

File: IFC4x3/Properties/m/MitigatedRiskLikelihood_2mh3cZyunB0ONymM2LqXn/DocProperty.xml (hunk 3 above)
Comment (jmirtsch) [reply IFC4x3/Properties/m/MitigatedRiskLikelihood_2mh3cZyunB0ONymM2LqXn/DocProperty.xml:2]: This one also seems to have had a duplicate, the property preserved in the main branch has a $ at the end. Note within an xml href id, the $ character is invalid and suppressed within the file (but not necessarily the folder, perhaps we should review this?)  https://github.com/bSI-InfraRoom/IFC-Specification/tree/64f1fc0e8cdf8aba82d2b99a423ddefae2f5a2ba/IFC4x3/Properties/m

File: IFC4x3/Properties/m/MitigatedRiskLikelihood_2mh3cZyunB0ONymM2LqXn/Documentation.md (hunk 4 above)
Comment (jmirtsch) [reply IFC4x3/Properties/m/MitigatedRiskLikelihood_2mh3cZyunB0ONymM2LqXn/Documentation.md:1]: Refer above about duplicate.

File: IFC4x3/PropertyConstants/v/VASCO COSTA_1jQ7XYFLbCQggk_8EEnXx4.xml (hunk 7 above)
Comment (jmirtsch) [reply IFC4x3/PropertyConstants/v/VASCO COSTA_1jQ7XYFLbCQggk_8EEnXx4.xml:2]: Seems to be another ghost duplicate, https://github.com/bSI-InfraRoom/IFC-Specification/tree/64f1fc0e8cdf8aba82d2b99a423ddefae2f5a2ba/IFC4x3/PropertyConstants/v

File: IFC4x3/Properties/n/NominalDepth_1shYs0qXWHuO00025QrE$V/DocProperty.xml (hunk 5 above)
Comment (jmirtsch) [reply IFC4x3/Properties/n/NominalDepth_1shYs0qXWHuO00025QrE$V/DocProperty.xml:5]: Yes, I have copied the localization in subsequent commit.

File: IFC4x3/Properties/n/NominalWidth_2bcxE0qSOHuO00025QrE$V/DocProperty.xml (hunk 6 above)
Comment (jmirtsch) [reply IFC4x3/Properties/n/NominalWidth_2bcxE0qSOHuO00025QrE$V/DocProperty.xml:5]: Yes, I have copied the localization in subsequent commit.

File: IFC4x3/Properties/d/Depth_19jxXCHUr5VhPM7hPVeFYR/DocProperty.xml (hunk 8 above)
Comment (jmirtsch) [reply IFC4x3/Properties/d/Depth_19jxXCHUr5VhPM7hPVeFYR/DocProperty.xml:3]: Yes, any property without a href id is orphaned.

File: IFC4x3/Sections/Domain specific data schemas/Schemas/IfcRail/PropertySets/Pset_RailwayAlignmentCommon/DocPropertySet.xml (hunk 9 above)
Comment (jmirtsch) [reply IFC4x3/Sections/Domain specific data schemas/Schemas/IfcRail/PropertySets/Pset_RailwayAlignmentCommon/DocPropertySet.xml]: This has been done in a subsequent commit.

File: IFC4x3/Properties/d/CargoDeadWeight_2vVxFsGGb3QOAIX9UEyhga/DocProperty.xml (hunk 1 above)
Comment (AlexBrad1eyCT) [reply IFC4x3/Properties/d/CargoDeadWeight_2vVxFsGGb3QOAIX9UEyhga/DocProperty.xml:2]: my i see it is fine now!

File: IFC4x3/Properties/m/MitigatedRiskLikelihood_2mh3cZyunB0ONymM2LqXn/DocProperty.xml (hunk 3 above)
Comment (AlexBrad1eyCT) [reply IFC4x3/Properties/m/MitigatedRiskLikelihood_2mh3cZyunB0ONymM2LqXn/DocProperty.xml:2]: It does need addressing, I do remember this $ i tried removing it but kept on getting put back by IFC doc if you've sorted it then good.

File: IFC4x3/PropertyConstants/v/VASCO COSTA_1jQ7XYFLbCQggk_8EEnXx4.xml (hunk 7 above)
Comment (AlexBrad1eyCT) [reply IFC4x3/PropertyConstants/v/VASCO COSTA_1jQ7XYFLbCQggk_8EEnXx4.xml:2]: I see it is correct.

**Review by AlexBrad1eyCT — APPROVED** [review]
Thanks for the Updates, looks all good to me!

**Pull request merged**